test/unit/test_file_download.py

```diff
@@ -0,0 +1,104 @@
+"""Tests for file downloading utilities"""
+from unittest import TestCase
+from unittest.mock import patch, mock_open, MagicMock, call
+
+import requests_mock
+from requests import HTTPError
+
+from terrawrap.utils.file_download import download_file, download_plugins
+
+
+class TestFileDownload(TestCase):
+    """Tests for file downloading utilities"""
+    @requests_mock.Mocker()
+    @patch('builtins.open', new_callable=mock_open)
+    @patch('os.path.isfile', MagicMock(return_value=True))
+    def test_file_download(self, mock_requests, open_mock):
+        """Test downloading a file"""
+        mock_requests.register_uri('GET', 'http://example.com', content=b'fake content')
+
+        download_file('http://example.com', '/tmp/plugins/foo')
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Probable insecure usage of temp file/directory.](https://app.codacy.com/gh/amplify-education/terrawrap/pullRequest?prid=5656512)